hal/src/nRF52840/usb_hal_cdc.c

```diff
@@ -87,6 +83,9 @@ typedef struct {
     volatile bool           rx_done;
 
     void (*bit_rate_changed_handler)(uint32_t bitRate);
+    HAL_USB_State_Callback  state_callback[MAX_USB_STATE_CB_NUM];
+    void*                   state_callback_context[MAX_USB_STATE_CB_NUM];
+    int                     state_callback_count;
```

Comment:
This seems unnecessary. The unused slots should be `NULL`.

Reply:
`state_callback_count` is used as the index of the callback slot when registering or calling the callback.

Reply:
But why? You can simply traverse the array, find a `NULL` entry and set replace it with the callback.

Reply:
OK, got it.